Marketplace: Add CollectionListView

lsl opened the pull request:
#### Proposed Changes

* Adds collection list view component #68800 #67987
* Adds some hard coded collections #67925 #67980 #67982 - Note missing some per [this comment](https://github.com/Automattic/wp-calypso/issues/67987#issuecomment-1274118347). We can add more later.
* Shrinks the compact plugin list view to 3 columns based on figma f84fqtCG4pCnCh2Hq0But5-fi-3620%3A32644 #67979
* Drops the short list constant to 4 from 6 to match figma #68877

#### Testing Instructions

* Plugins screen should render the new collections sections
* Old sections should be shrunk to 4 from 6

Fixes #68800
Fixes #67979
Fixes #67925
Fixes #67980
Fixes #67982
Fixes #68877

Related to #67987

Before 
![Screenshot 2022-10-11 at 17-09-50 Plugins ‹ Bussiness Simple — WordPress com](https://user-images.githubusercontent.com/811776/195009890-bde6bc51-f72e-4e12-b5e3-3fce6278ee3d.png)

After
![Screenshot 2022-10-11 at 15-29-10 Plugins ‹ Simple 123 — WordPress com](https://user-images.githubusercontent.com/140841/195192211-bbf88362-0e39-447d-9a99-97a8c8f4759f.png)

lsl committed:
Add CollectionsListView

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7575186

Thank you @lsl for including a screenshot in the description! This is really helpful for our translators.

lupus2k committed:
Removed 2 plutins from Featured because now only shows 4 plugins

lupus2k committed:
Removed commented featured tests

allilevine commented:
@vinimotaa Adding [your feedback](https://github.com/Automattic/wp-calypso/issues/67979#issuecomment-1274847665) here to address:

> Hi @lsl, thanks for this, is this ready for a design review? Product cards seem a bit off from the design specs provided. If the team is still working on it, it's ok, otherwise, we could double-check check paddings, font sizes, thumbnail sizes, etc. Happy to help

@lsl had also [asked here about icons.](https://github.com/Automattic/wp-calypso/issues/67987#issuecomment-1274118347)

DustyReagan committed:
Update plugins-browser-item.compact CSS

DustyReagan commented:
> Product cards seem a bit off from the design specs provided

@vinimotaa I think I see what you're referring to. I updated the CSS to decrease the size of the compact icon to 40px, and adjusted the margin a bit to suit the smaller icon size. Please let us know if there's more you see. ☺️

DustyReagan reviewed (approved):
Yoast is a bit over-represented, and we're still waiting on some translations, but otherwise, it looks good to me! 👍

paulopmt1 committed:
Improved media-query breaks at CSS:
- Default css for mobile
- Tablet with only two blocks
- Wide screen with three blocks
- Small sizes with text-overflow: ellipsis

paulopmt1 reviewed (approved):
Good job, everyone! Agree with @DustyReagan, tomorrow seems to be a good day to merge it so we'll have more translations in place and probably a replacement for Yoast in the monetization category.

lsl committed:
Typescript fix after rebase

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7575186).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7575186).
